tests/run-make/rustdoc-io-error/rmake.rs

```diff
@@ -0,0 +1,26 @@
+// This test verifies that rustdoc doesn't ICE when it encounters an IO error
+// while generating files. Ideally this would be a rustdoc-ui test, so we could
+// verify the error message as well.
+//
+// It operates by creating a temporary directory and modifying its
+// permissions so that it is not writable. We have to take special care to set
+// the permissions back to normal so that it's able to be deleted later.
```

Comment:
The comment says
> We have to take special care to set
> the permissions back to normal so that it's able to be deleted later.
but AFAICT the permission is not restored?

Reply:
Good point, fixing that.